prov/efa/src/efa_cq.c

```diff
@@ -69,8 +69,8 @@ ssize_t efa_cq_readerr(struct fid_cq *cq_fid, struct fi_cq_err_entry *entry,
 
 	entry->op_context = (void *)(uintptr_t)cq->ibv_cq_ex->wr_id;
 	entry->flags = efa_cq_opcode_to_fi_flags(ibv_wc_read_opcode(cq->ibv_cq_ex));
-	entry->err = EIO;
-	entry->prov_errno = cq->ibv_cq_ex->status;
+	entry->err = FI_EIO;
+	entry->prov_errno = ibv_wc_read_vendor_err(cq->ibv_cq_ex);
```

Comment:
Please run unit tests to double check backward compatibility, or modify the broken tests accordingly.

Reply:
It might also be a good idea to assert on the prov error in unit tests as well - that should hit this code path.

Reply:
It looks like this goes to a function pointer for read_vendor_error in the ibv_cq_ex struct. Are you sure this can't be a NULL function?

Reply:
Hmm, I'm not 100% sure. I'm inferring from the existing `cq->ibv_cq_ex->status` dereference on L65, that the `ibv_cq_ex` struct was properly initialized via `ibv_create_cq_ex()`.